www-apps/redmine/redmine-3.3.3.ebuild

```diff
@@ -0,0 +1,218 @@
+IUSE="imagemagick fastcgi ldap markdown passenger"
+
+# nokogiri ruby 2.1 >= 1.7.0, else 1.6.8
+
+ruby_add_rdepend "
+	dev-ruby/actionpack-action_caching
+	dev-ruby/actionpack-xml_parser:*
+	>=dev-ruby/builder-3.2.2:3.2
+	>=dev-ruby/coderay-1.1.0
+	dev-ruby/i18n:0.7
+	>=dev-ruby/rails-html-sanitizer-1.0.3
+	>=dev-ruby/rack-openid-0.2.1
+	>=dev-ruby/jquery-rails-3.1.4:3
+	dev-ruby/loofah
+	dev-ruby/mime-types:*
+	dev-ruby/mimemagic
+	>=dev-ruby/nokogiri-1.6.8
+	dev-ruby/protected_attributes
+	>=dev-ruby/rails-4.2.5.2:4.2
+	dev-ruby/request_store:1.0.5
+	>=dev-ruby/roadie-rails-1.1.0
+	>=dev-ruby/rbpdf-1.19.2
+	>=dev-ruby/ruby-openid-2.3.0
+	dev-ruby/rubygems
+	fastcgi? ( dev-ruby/fcgi )
+	imagemagick? ( >=dev-ruby/rmagick-2.14.0 )
+	ldap? ( >=dev-ruby/ruby-net-ldap-0.12.0 )
+	markdown? ( >=dev-ruby/redcarpet-3.3.2 )
+	passenger? ( www-apache/passenger )
+	"
+
+#ruby_add_bdepend ">=dev-ruby/rdoc-2.4.2
+#	dev-ruby/yard
+#	test? (
+#		>=dev-ruby/shoulda-3.3.2
+#		>=dev-ruby/mocha-0.13.3
+#		>=dev-ruby/capybara-2.0.0
+#		<dev-ruby/nokogiri-1.6.0
+#	)"
+
+REDMINE_DIR="/var/lib/${PN}"
+
+pkg_setup() {
+	enewgroup redmine
+	enewuser redmine -1 -1 "${REDMINE_DIR}" redmine
+}
+
+all_ruby_prepare() {
+	rm -r log files/delete.me Gemfile || die
+
+	# bug #406605
+	rm .{git,hg}ignore || die
+
+	cat > ${T}/50${PN} <<-EOF
+		CONFIG_PROTECT="${EROOT%/}${REDMINE_DIR}/config"
+		CONFIG_PROTECT_MASK="${EROOT%/}${REDMINE_DIR}/config/locales ${EROOT%/}${REDMINE_DIR}/config/settings.yml"
+	EOF
+
+	# remove ldap staff module if disabled to avoid #413779
+	use ldap || rm app/models/auth_source_ldap.rb || die
+
+	epatch "${FILESDIR}/${P}_requires.patch"
+}
+
+all_ruby_install() {
+	dodoc doc/{CHANGELOG,INSTALL,README_FOR_APP,RUNNING_TESTS,UPGRADING}
+	dodoc README.rdoc
+	rm -r doc appveyor.yml CONTRIBUTING.md README.rdoc || die
+
+	keepdir /var/log/${PN}
+
+	insinto "${REDMINE_DIR}"
+	doins -r .
+	insinto "${REDMINE_DIR}/config"
+	doins "${FILESDIR}/additional_environment.rb"
+	keepdir "${REDMINE_DIR}/files"
+	keepdir "${REDMINE_DIR}/public/plugin_assets"
+
+	fowners -R redmine:redmine \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/public/plugin_assets" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	fowners redmine:redmine "${REDMINE_DIR}"
+
+	# protect sensitive data, see bug #406605
+	fperms -R go-rwx \
+		"${REDMINE_DIR}/config" \
+		"${REDMINE_DIR}/files" \
+		"${REDMINE_DIR}/tmp" \
+		/var/log/${PN}
+
+	if use passenger; then
+		has_apache
+		insinto "${APACHE_VHOSTS_CONFDIR}"
+		doins "${FILESDIR}/10_redmine_vhost.conf"
+	fi
+
+	# logrotate
+	insinto /etc/logrotate.d
+	newins "${FILESDIR}"/redmine.logrotate redmine
+
+	newconfd "${FILESDIR}/${PN}.confd" ${PN}
+	newinitd "${FILESDIR}/${PN}-3.initd" ${PN}
+	doenvd "${T}/50${PN}"
+}
+
+pkg_postinst() {
+	einfo
+	if [ -e "${EROOT%/}${REDMINE_DIR}/config/initializers/session_store.rb" ] || [ -e "${EROOT%/}${REDMINE_DIR}/config/initializers/secret_token.rb" ]; then
```

Comment:
Use [[ ... ]] in bash, and then you can put || inside. Also, line wrapping would probably be beneficial to the reader ;-).